### ISSUE TITLE
Does not handle active/inactive tabs appropriately

### ISSUE THREAD

**Issue by theadam** [paldepind/flyd-every#1]
When in an inactive tab, timeouts are fired at most once per second.  [See this document](https://developer.mozilla.org/en-US/docs/Web/API/WindowTimers/setTimeout#Inactive_tabs).  

Due to the way the next timeout is calculated inside flyd-every, this action can cause problems.  This is not the easiest thing to recreate but [here is some code to try out](http://requirebin.com/?gist=c690ce93158a6294096d).  Open your console, watch the events being logged.  Change to another tab for a while (the longer the better, but even a few seconds will be enough to witness the issue).  Once you change back to the running application, the stream will fire many events in a row.  The setTimeout calls were being called at most one second, but the math was expecting them to be called every 500ms.  This causes setTimeout to be called with negative numbers.

**Comment by paldepind** [comment]
Yes. This is an an issue. Are you facing this problem in actual code? The only alternative I can see to the firing of many events in a row would be to drop events.

I choose the current behavior because `every` promises to emit a value every duration specified. And when the `setTimeout` isn't being called often enough the only way to get as many calls as requested is to emit a lot of values really fast.

``` javascript
flyd.scan(function(acc, _) {
  console.log('I began ' + (acc+1) + 'seconds ago');
  return acc + 1;
}, 0, everySecond);
```

Even if switching tabs the above would always stabilize at a correct count. If dropping events it would turn incorrect.

I don't think calling `setTimeout` with a negative number causes any problems.

**Comment by theadam** [comment]
I see, if this is the intended behavior then of course it should stay.  I was just messing around and using this to fire events each frame for a game (I have since switched to a stream that fires using request animation frame).  I noticed that when switching tabs and coming back to the game, my snake would crash into walls at high speeds.

It sounds more like I am just using flyd-every where something else would be more appropriate :).  Thanks again!

**Comment by paldepind** [comment]
If you want to do something on each frame then using `requestAnimationFrame` is appropriate.

If you want to do something like spawn an enemy every 5 seconds then flyd-every does the job. And in that case you don't want the user to prevent enemy spawning by switching frame and coming back.

But maybe keeping a lower limit on how fast the stream emits would be appropriate? Like, if a user request an update every second then never emit faster than half a second to keep up.

**Comment by theadam** [comment]
I think that is a good idea
